lib/src/Base/Graph/DrawableImplementation.cxx

```diff
@@ -924,6 +927,64 @@ Point DrawableImplementation::ConvertFromHSVIntoRGB(const Scalar hue,
   return redGreenBlue;
 }
 
+/* Convert a RGB triplet to HSV */
+Point DrawableImplementation::ConvertFromRGBIntoHSV(const Scalar red,
+    const Scalar green,
+    const Scalar blue)
+{
+  if (red < 0.0 || red > 1.0) throw InvalidArgumentException(HERE) << "Invalid red="<<red;
+  if (green < 0.0 || green > 1.0) throw InvalidArgumentException(HERE) << "Invalid green="<<green;
+  if (blue < 0.0 || blue > 1.0) throw InvalidArgumentException(HERE) << "Invalid blue="<<blue;
+  return ConvertFromRGBIntoHSV(static_cast<UnsignedInteger>(round(255 * red)),
+                               static_cast<UnsignedInteger>(round(255 * green)),
+                               static_cast<UnsignedInteger>(round(255 * blue)));
+}
+/* Convert a RGB triplet to HSV */
+Point DrawableImplementation::ConvertFromRGBIntoHSV(const UnsignedInteger red,
+    const UnsignedInteger green,
+    const UnsignedInteger blue)
+{
+  if (red > 255) throw InvalidArgumentException(HERE) << "Invalid red="<<red;
+  if (green > 255) throw InvalidArgumentException(HERE) << "Invalid green="<<green;
+  if (blue > 255) throw InvalidArgumentException(HERE) << "Invalid blue="<<blue;
+  const Scalar redScalar(red);
+  const Scalar greenScalar(green);
+  const Scalar blueScalar(blue);
+  const Scalar maxColor = std::max(std::max(redScalar, greenScalar), blueScalar);
+  const Scalar minColor = std::min(std::min(redScalar, greenScalar), blueScalar);
+  const Scalar chroma = maxColor - minColor;
+  Scalar hue, saturation, value;
+  
+  if(chroma > 0) {
+    if(maxColor == redScalar) {
+      hue = 60.0 * ((greenScalar - blueScalar) / chroma);
+    } else if(maxColor == greenScalar) {
+      hue = 60.0 * ((blueScalar - redScalar) / chroma + 2.0);
+    } else {
+      // maxColor == blue
+      hue = 60.0 * ((redScalar - greenScalar) / chroma + 4.0);
+    }
+    if(maxColor > 0.0) {
+      saturation = chroma / maxColor;
+    } else {
+      saturation = 0.0;
+    }
+    value = maxColor / 255.0;
+  } else {
+    hue = 0.0;
+    saturation = 0.0;
+    value = maxColor / 255.0;
+  }
+  if(hue < 0.0) {
+    hue += 360.0;
+  }
+  Point hsv(3);
+  hsv[0] = hue;
+  hsv[1] = saturation;
+  hsv[2] = value;
+  return hsv;
```

Comment:
a neat thing is braces initializer: `const Point hsv = {hue, saturation, value};`

Reply:
Much clearer indeed. Fixed.